packages/plugins/geth/src/devtxs.ts

```diff
@@ -4,18 +4,24 @@ import Web3 from "web3";
 import constants from "embark-core/constants.json";
 export default class DevTxs {
   private embark: Embark;
+  private blockchainConfig: any;
   private events: Events;
   private logger: Logger;
   private web3?: Web3;
   private regularTxsInt?: NodeJS.Timeout;
   constructor(embark: Embark) {
     this.embark = embark;
+    this.blockchainConfig = this.embark.config.blockchainConfig;
     this.logger = embark.logger;
     this.events = embark.events;
   }
   public async init() {
-    const provider = await this.events.request2("blockchain:client:provider", "ethereum");
-    this.web3 = new Web3(provider);
+    const { rpcHost, rpcPort, wsRPC, wsHost, wsPort } = this.blockchainConfig;
+    if (wsRPC) {
+      this.web3 = new Web3('ws://' + wsHost + ':' + wsPort);
+    } else {
+      this.web3 = new Web3(rpcHost + ':' + rpcPort);
+    }
```

Comment:
It would be good if we still used `blockchain:client:provider` (sets WS headers and fragmentationThreshold) as well as `buildUrl` (which is unit tested) to consistently build the endpoint URL.
```
const { rpcHost, rpcPort, wsRPC, wsHost, wsPort } = this.blockchainConfig;
let endpoint;
if (wsRPC) {
  endpoint = buildUrl("ws", wsHost, wsPort, "ws");
} else {
  endpoint = buildUrl("http", rpcHost, rpcHost, "rpc");
}

const provider = await this.events.request2("blockchain:client:provider", "ethereum", endpoint);

this.web3 = new Web3(provider);
```

Reply:
Actually, not need to get the parts at all. The blockchain config now has an `endpoint` parameter that contains the built URL already.
So you can just connect to `this.blockchainConfig.endpoint`
You can still follow the trick Eric said about using the request anyway though.